src/Microsoft.Azure.WebJobs.Extensions.Kafka/Listeners/KafkaTargetScaler.cs

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Confluent.Kafka;
+using Microsoft.Azure.WebJobs.Host.Scale;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Kafka
+{
+    public class KafkaTargetScaler<Tkey, TValue> : ITargetScaler
```

Comment:
The class should not be public as we want to keep the internal implementation.

Reply:
Since we're returning an instance of this class within a public method (`ITargetScaler GetTargetScaler()` returns it) we cannot restrict the accessibility of methods present within it to internal assembly only. This would restrict all its methods to internal. Even public methods like `GetScaleResultAsync()` will be restricted to internal, removing access for Webjobs SDK to call on this method to retrieve target scaling result. To avoid this, class is made public although the constructor to this class is internal - meaning other packages do not have access to creating a new instance.

Reply:
Updated.